File: php/features.php
<?php
declare(strict_types=1);

// ThrowawayEmail SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ThrowawayEmailFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ThrowawayEmailBaseFeature();
            case "test":
                return new ThrowawayEmailTestFeature();
            default:
                return new ThrowawayEmailBaseFeature();
        }
    }
}

File: py/features.py
# ThrowawayEmail SDK feature factory

from feature.base_feature import ThrowawayEmailBaseFeature
from feature.test_feature import ThrowawayEmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ThrowawayEmailBaseFeature(),
        "test": lambda: ThrowawayEmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
